test/Rentals.spec.ts

```diff
@@ -1338,6 +1431,81 @@ describe('Rentals', () => {
+      const rentalDays2 = 100
+
+      let rental = await rentals.rentals(listingParams.contractAddress, listingParams.tokenId)
+
+      expect(rental.lessor).to.equal(zeroAddress)
+      expect(rental.tenant).to.equal(zeroAddress)
+      expect(rental.endDate).to.equal(0)
+
+      await network.provider.send('evm_setAutomine', [false])
+
+      offerParams = { ...offerParams, rentalDays: rentalDays1 }
+
+      await rentals.connect(lessor).acceptOffer({ ...offerParams, signature: await getOfferSignature(tenant, rentals, offerParams) })
+
+      offerParams = { ...offerParams, rentalDays: rentalDays2, nonces: [0, 0, 1] }
+
+      await rentals.connect(lessor).acceptOffer({ ...offerParams, signature: await getOfferSignature(tenant, rentals, offerParams) })
+
+      await network.provider.send('evm_setAutomine', [true])
+
+      await evmMine()
+
+      const latestBlockTimestamp = await getLatestBlockTimestamp()
+
+      rental = await rentals.rentals(listingParams.contractAddress, listingParams.tokenId)
+
+      expect(rental.lessor).to.equal(lessor.address)
+      expect(rental.tenant).to.equal(tenant.address)
+      expect(rental.endDate).to.equal(latestBlockTimestamp + daysToSeconds(rentalDays1 + rentalDays2))
+    })
+
+    it('should allow calling accept offer then an accept listing if the second is an extension in the same block', async () => {
+      const rentalDays1 = 10
+      const rentalDays2 = 100
+
+      let rental = await rentals.rentals(listingParams.contractAddress, listingParams.tokenId)
+
+      expect(rental.lessor).to.equal(zeroAddress)
+      expect(rental.tenant).to.equal(zeroAddress)
+      expect(rental.endDate).to.equal(0)
+
+      await network.provider.send('evm_setAutomine', [false])
+
+      offerParams = { ...offerParams, rentalDays: rentalDays1 }
+
+      await rentals.connect(lessor).acceptOffer({ ...offerParams, signature: await getOfferSignature(tenant, rentals, offerParams) })
+
+      listingParams = { ...listingParams, maxDays: [rentalDays2], minDays: [rentalDays2], nonces: [0, 0, 1] }
+      acceptListingParams = { ...acceptListingParams, rentalDays: rentalDays2, index: 0 }
+
+      await rentals
+        .connect(tenant)
+        .acceptListing(
+          { ...listingParams, signature: await getListingSignature(lessor, rentals, listingParams) },
+          acceptListingParams.operator,
+          acceptListingParams.index,
+          acceptListingParams.rentalDays,
+          acceptListingParams.fingerprint
+        )
+
+      await network.provider.send('evm_setAutomine', [true])
+
+      await evmMine()
+
+      const latestBlockTimestamp = await getLatestBlockTimestamp()
+
+      rental = await rentals.rentals(listingParams.contractAddress, listingParams.tokenId)
+
+      expect(rental.lessor).to.equal(lessor.address)
+      expect(rental.tenant).to.equal(tenant.address)
+      expect(rental.endDate).to.equal(latestBlockTimestamp + daysToSeconds(rentalDays1 + rentalDays2))
+    })
```

Comment:
I would check again the nonces and rentals properties before and after each tx

Reply:
Done